Add `clusterid`, an optional parameter to storageclass.

humblec opened the pull request:
At present, admin doesn't have the privilege to chose the
trusted storage pool from which persistent gluster volume
has to be provided.

This patch introduce a new storage class parameter which allows
the admin to specify storage pool/cluster if required.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

<!-- Reviewable:start -->

---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36437)
<!-- Reviewable:end -->

zmerlynn commented:
@k8s-bot ok to test

jsafrane commented on pkg/volume/glusterfs/glusterfs.go:
```diff
@@ -369,6 +370,7 @@ type provisioningConfig struct {
 	secretNamespace string
 	secretName      string
 	secretValue     string
+	clusterid       string
```
camel case please, i.e. `clusterId`

humblec replied:
Done.

jsafrane commented on pkg/volume/glusterfs/glusterfs.go:
```diff
@@ -66,6 +66,7 @@ const (
 	annGlusterSecretNamespace = "glusterfs.kubernetes.io/secretnamespace"
 	annGlusterUserKey         = "glusterfs.kubernetes.io/userkey"
 	annGlusterUser            = "glusterfs.kubernetes.io/userid"
+	annGlusterClusterId       = "glusterfs.kubernetes.io/clusterid"
```
This annotation is not used anywhere, please remove it.
[Also, if you like, you can trim the other unused annotations too]

humblec replied:
Done.

jsafrane commented on pkg/volume/glusterfs/glusterfs.go:
```diff
@@ -510,7 +512,8 @@ func (p *glusterfsVolumeProvisioner) CreateVolume() (r *api.GlusterfsVolumeSourc
 		return nil, 0, fmt.Errorf("failed to create gluster REST client, REST server authentication failed")
 	}
-	volumeReq := &gapi.VolumeCreateRequest{Size: sz, Durability: gapi.VolumeDurabilityInfo{Type: durabilityType, Replicate: gapi.ReplicaDurability{Replica: replicaCount}}}
+	clusterids := dstrings.Split(p.clusterid, ",")
```
camel xase please

humblec replied:
Done

humblec commented:
@jsafrane Thanks ! Addressed your comments and also rebased the patch.

jsafrane commented:
/lgtm for 1.6

humblec commented:
@jsafrane Rebased the patch to master.

jsafrane commented:
/lgtm

humblec committed:
At present, admin doesn't have the privilege to choose the
trusted storage pool from which persistent gluster volume
has to be provided.

This patch introduce a new storage class parameter which allows
the admin to specify storage pool/cluster if required.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec commented:
@jsafrane once again rebased :) , it removed LGTM.

jsafrane commented:
/lgtm

jsafrane commented:
/lgtm
!!!

jsafrane commented:
@k8s-bot kubemark e2e test this
